modules/page_base.py

```diff
@@ -711,3 +711,29 @@ def loaded(self):
             pass
         self.set_content_context()
         return _loaded
+
+    def get_css_zoom(self):
+        """
+        Checks the CSS zoom and transform scale to determine the current zoom level.
+        """
+        # Retrieve the CSS zoom property on the body element
+        css_zoom = self.driver.execute_script("return window.getComputedStyle(document.body).zoom")
+        if css_zoom:
+            return float(css_zoom)
+
+        # If zoom property is not explicitly set, check the transform scale
+        css_transform_scale = self.driver.execute_script("""
+            const transform = window.getComputedStyle(document.body).transform;
+            if (transform && transform !== 'none') {
+                return transform;
+            } else {
+                return null;
+        """)
+
+        # Parse the transform matrix to extract the scale factor (e.g., matrix(a, b, c, d, e, f))
+        if css_transform_scale:
+            scale_factor = float(css_transform_scale.split('(')[1].split(',')[0])
+            return scale_factor
+
+        # Default return if neither zoom nor transform is set
+        return 1.0
```

Comment:
you could add another method here or in the BOM that does something like (code not guaranteed to work):

```Python
def wait_for_zoom(self, factor: float) -> Page:
    self.expect(lambda _: self.get_css_zoom) == factor)
    return self
```